# Introduction

P-values have been the reason behind lack of reproducibility in scientific discoveries and especially in replicated studies and multiple testing. This problem is especially prevalent in Genome-Wide Association Studies (GWAS), where estimated effects have upward bias and often fail to replicate in validation studies. This phenomenon is known as the winner’s curse  [@zollner2007overcoming]. To account for this discrepancy, previous studies perform two analyses: one with all the data together, and one only using the validation site data. However, this approach is based on the underlying assumption that the association found in discovery sites is true, which is problematic for multiple testing applications such as genome-wide association studies. Furthermore, if there is a true effect, leaving out the discovery data, which could be a large portion of the total dataset, reduces power.

One example of the winner’s curse in action is the analysis of the association between single nucleotide polymorphisms (SNPs) in the p53 protein, which is needed for cell growth and DNA repair, and invasive ovarian cancer. Mixed effect SNP-at-a-time analysis of 5 SNPs that were chosen for replication resulted in associations between 2 SNPs and serous invasive cancer [@Schildkraut2349]. Only one of these was strongly supported to be associated in a follow-up analysis using Multi-level Inference for SNP Association (MISA), which employs Bayesian Model Averaging and Bayes Factors for selection [@schildkraut2010association]. However, most recent studies have not found evidence of association between any TP53 SNPs and cancer [@phelan2017identification].

After a review of existing literature dealing with this issue, we propose three approaches to adress this: a fully Bayesian model, a conditional likelihood prior for discovery site data, and a bayes factor approximation to the probability of association. The performance of these methods is tested on normal simulations, and then on hierarchical simulations split into discovery and validation "sites". All three proposed methods provide improvements over naive models. Finally, the models are used to reanalyze the p53 SNPs; none of the SNPs were found to be significant under any model.

