 <!--
Notes on articles.
-->

# Literature Review  {#lit-review}

## Background and Motivation
    
#### Redefine statistical significance
Benjamin et al 2017
https://www.nature.com/articles/s41562-017-0189-z

- map Bayes factors to p-values
    - depends on H1 so it needs to be for specific alternatives
    - plots of bayes factors to p values show that 5% threshold is too high
    - suggests .005 to reduce false positives (esp in low power)
    

#### Calibration of p Values for Testing Precise Null Hypotheses
Sellke et al 2003
http://www.tandfonline.com/doi/pdf/10.1198/000313001300339950?needAccess=true

- transform p value to bayes factor (lower bound on this) and alpha (type 1 error)

---

## Frequentist Methods

### Conditional Likelihood

#### Overcoming the Winner’s Curse: Estimating Penetrance Parameters from Case-Control Data

Zollner and Pritchard, 2007

https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1852705/

- instead of calculating odds-ratio, use prevalance to calculate penetrance 

- likelihood conditional on parameters and significance at level alpha.
- algorithm:
    - sample from L($\theta, \phi$) (population freq and penetrance)
    - get L(data | phi, theta) and pick highest
    - perturb each param by epsilon, if better accept
    - to test, generate 3 samples: naive, corrected, and estimated w an independent sample
  

#### Estimating odds ratios in genome scans: an approximate conditional likelihood approach.

Ghosh A, Zou F, Wright FA 2008

https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2665019/

- proposes mean of conditional likelihood, MLE, and mixture of both 
- there used to be code online www.bios.unc.edu/~wrightf/genomebias/

#### Bias-reduced estimators and confidence intervals for odds ratios in genome-wide association studies

Zhong H, Prentice RL. 2008

- similar to Zollner and Pritchard, but with asymptotic approximation instead of sampling
- "slightly different" likelihood formulation

#####Statistical correction of the Winner’s Curse explains replication variability in quantitative trait genome-wide association studies
Palmer and Pe'er. 2017

- based on Zhong and Prentice.
- code here: https://github.com/cpalmer718/gwas-winners-curse

###Bootstrap

####BR-squared: a practical solution to the winner’s curse in genome-wide scans
Sun et al. 2011 
https://www.ncbi.nlm.nih.gov/pubmed/21246217

- bootstrapping to address ranking effect and threshold effect
- independent detection and validation/effect estimation using sample split
- calculate effect for both in and out of bootstrap sample x, then get bias as diff and adjust
- can use genome wide bootstrap to improve over likelihood based methods
- software here: http://www.utstat.toronto.edu/sun/Software/BR2/

###Shrinkage

####Extracting Actionable Information From Genome Scans

Bacanu and Kendler. 2012

http://onlinelibrary.wiley.com/doi/10.1002/gepi.21682/full

- soft threshold, adjusts statistics such that their sum of squares do not overestimate the true mean

#### A simple yet accurate correction for winner's curse can predict signals discovered in much larger genome scans

Bigdelli et al. 2016

https://academic.oup.com/bioinformatics/article/32/17/2598/2450747

- uses multiple testing adjustment (False Discovery Rate) FDR Inverse Quantile Transformation (FIQT)
- transforms to p vals, shrinks towards one, and then transforms back to z scores
- code: https://github.com/bacanusa/FIQT

#####Statistical significance for genomewide studies

Storey and Tibshirani. 2003

http://www.pnas.org/content/100/16/9440.full

- FDR = expectation of false positives/significant features
- q is "proportion of false positives incurred when calling that feature significant"
    - sort of Bayesian (probability that a feature is null given it is significant)
- provides estimates for proportion of truly null features empirically
    - there's several other papers on how to estimate this


## Bayesian Methods

### "Fully Bayesian"

#### Bayesian methods to overcome the winner's curse in genetic studies

Xu et al 2011

https://arxiv.org/pdf/0907.2770.pdf

- Bayesian framework with model averaging
- logistic regression example with biased estimator, approximated with normal by CLT
- (hierarchical) Bayesian bias correction
    - spike-and-slab prior for mu, p(mu given epsilon)
    - Beta prior for epsilon p(epsilon given a,b)
      -unifor prior is problematic because of Bartlett's paradox
    - inv gamma for sigma sq with mean  = s^2 and var = 200
        - 200 chosen bc empirically doesn’t matter much
    - sample from posterior w data augmentation metropolis hasting
- BMA without conditioning on significance
    
### Empirical Bayes 

#### Empirical Bayes Correction for the Winner's Curse in Genetic Association Studies

Ferguson et al 2013

https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4048064/

- prior of mean mu_i from distribution f
- empirical bayes method
    - find posterior (where f is the prior) and use the mean
        - results in smallest sq error
    - f is unknown, but can express mean and variance of posterior with marginal p(z)
        - can estimate empirical log(p(z)) with binning and then fitting splines
          - nonparametric but still dependent on spline function, bins, etc.
        - can achieve pseudo Bayesian credible intervals
          - these are not appropriate though, because they are symmetric (which is not guaranteed for multimodal or asymmetric distributions)
          - would be better to use HPD
    - combine with cond likelihood to account for tail behavior by comparing length of 95%CI for empirical bayes and conditional likelihood intervals, and pick the shortest
- simulation study with data from 3 distributions (normal, contamindated normal, double exponential) to test robustness

#### Power estimation and sample size determination for replication studies of genome-wide association studies

Jiang and Yu. 2016

https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-015-2296-4

- interested in power
  - "bayesian power" takes the expectation over mu of P(replication significant | original study, mu, H1)
- spike and slab prior
- hyperparameters estimated empirically
  - this will underestimate the variance
- proportion of null hypothesis is based on another parameter gamma corresponding to a p-value threshold
- can calculate Bayesian predictive power estimator 
- simulation to compare estimate to true power


### Adjusted Bayesian inference for selected parameters

Yekutieli. 2012

- necessary to have some adjustement for selection in bayesian framework
- false coverage instead of p val, etc
