# Application

```{r}
knitr::opts_chunk$set(echo = FALSE)
knitr::opts_chunk$set(eval = FALSE)
knitr::opts_chunk$set(cache = FALSE)

library(rmeta)
library(lme4)
library(R2jags)
library(xtable)
library(knitr)
library(dplyr)
library(ggplot2)
library(reshape2)

source("HPD.R")
```

## Genome Study

### Introduction

The aim is to study the association between single nucleotide polymorphisms (SNPs) in the p53 protein, which is needed for cell growth and DNA repair, and invasive ovarian cancer. Its role in carcinogenesis has been confirmed in various studies (for more detailed information, see **[p53 paper]**).

### Data

#### Discovery Sites

Three independent discovery studies focused on TP53 polymorphisms and risk of ovarian cancer: the North Carolina Ovarian Cancer Study (NCOCS),(21) the Mayo Clinic Case-Control Study (MAYO),(22) and the Polish Ovarian Cancer Study (POCS). These were restricted to non-Hispanic white women with newly diagnosed, histologically confirmed, primary invasive epithelial ovarian cancer and to non-Hispanic white controls. 23 SNPs were genotyped in total, with some overlap between sites.

```{r}
##table of cases and controls
```



#### Replication Sites

Ten other sites contributed data: the Australian Ovarian Cancer Study (AOCS) and the Australian Cancer Study (ACS) presented together as AUS, the Family Registry for Ovarian Cancer (FROC, presented as STA), the Hawaiian Ovarian Cancer Study (HAW), the Malignant Ovarian Cancer Study Denmark (MALOVA), the New England Case-Control Study (NEC), the Nurses' Health Study (NHS), SEARCH Cambridge (SEA), the Los Angeles County Case-Control Study of Ovarian Cancer (LAC-CCOC, presented here as USC), the University of California at Irvine study (UCI), and the United Kingdom Ovarian Cancer Population Study (UKOPS, presented here as UKO).

```{r}
##table of cases and controls
```

The combined data set (discovery and replication) comprised ** is this right, use inline r** 5,206 white, non-Hispanic invasive epithelial ovarian cancer cases, of which 2,829 were classified as serous invasive ovarian cancer, and 8,790 white non-Hispanic controls. 

### Motivation

Traditionally, genome-wide-significant associations often fail or have much smaller effects in replication tests because of the winner's curse and using discovery data twice (when testing for significance, and calculating the effect size). To account for this discrepancy, previous studies perform two analyses: one with all the data together, and one only using the validation site data. However, this approach not only throws away all the discovery data, which could be a large portion of the total dataset, reducing power, but also assumes that the significance found in the discovery data is certain, which is especially problematic in genome-wide association studies, which perform multiple testing to find associations.

We propose three different approaches: 

1. A Bayesian mixed effects hierarchical model that can jointly perform significance testing and effect estimation. By combining the testing and estimate steps, we can overcome the winner's curse and account for the uncertainty that arises when selecting an SNP(???).

2. A conditional likelihood model that can take into account the probability of finding a significant result in the discovery sites when estimating effect size.

3. A bayes factor based model that uses the bayes factor from the discovery sites (which is more reliable than the p-value, as discussed previously) to quantify the uncertainty of the significant result.

While the first approach is truly Bayesian and requires all the data, the second and third can be used as long as the sufficient statistics (MLE, SE, p-value, alpha) are available. 


### Bayesian Model

In this model, the prior distribution of $\mu$ is modeled with a spike and slab prior



```{r load data}
getdata<- function(snp.name, iter=5000, drop.sites= NULL){
  load("tp53.Rdata")
  if(!is.null(drop.sites)) {
    
    use = !(tp53epi.wsi$site %in% drop.sites)
  } else {
    use = rep(TRUE, nrow(tp53epi.wsi))
  }
  #this is a new indicator
  site.names = levels(factor(tp53epi.wsi[which(!is.na(tp53geno.wsi[,snp.name])),"site"]))
  discovery.sites <- which(site.names%in% drop.sites)
  
  p53.snp = tp53geno.wsi[use, snp.name]
  tp53epi.wsi = tp53epi.wsi[use,]
  
  missing.geno = is.na(p53.snp)
  site.names = levels(factor(tp53epi.wsi[!missing.geno,"site"]))
  
  p53.data = list(CaseCon=tp53epi.wsi$casecon[!missing.geno], site=as.numeric(factor(tp53epi.wsi$site[!missing.geno])), BC = as.numeric(tp53epi.wsi$prev.BC[!missing.geno]), Age = tp53epi.wsi$refage[!missing.geno], p53 = p53.snp[!missing.geno])
  
  p53.df = data.frame(p53.data)
  
  J = length(p53.data$CaseCon)
  n.sites = length(unique(p53.data$site))
  p53.data$J = J
  p53.data$n.sites = n.sites
  
  p53.data$discovery.sites<-discovery.sites
  p53.data$missing.geno<-missing.geno
  p53.data$site.names<-site.names
  
  return(p53.data)
  
}

```

```{r}
snp.name="rs12951053n" 
discovery.sitenames= c("POL", "MAY", "NCO")
#regular data
p53.data<- getdata(snp.name)
#validation data
p53.dataval<- getdata(snp.name, drop.sites=discovery.sitenames)

```




### EDA

- plot of data points?
- conditional likelihood, FDR, etc as function of Y
- 

### Frequentist Results
```{r frequentist analysis function}
OR.freq = function(snp.name, psdir="ps", drop.sites=NULL,iter=5000, debug=F) {
  load("tp53.Rdata")
  if (!is.null(drop.sites)) {
    use = !(tp53epi.wsi$site %in% drop.sites)}
  else {
    use = rep(TRUE, nrow(tp53epi.wsi))
  }
  
  p53.snp = tp53geno.wsi[use, snp.name]
  tp53epi.wsi = tp53epi.wsi[use,]
  
  missing.geno = is.na(p53.snp)
  site.names = levels(factor(tp53epi.wsi[!missing.geno,"site"]))
  
  p53.data = list(CaseCon=tp53epi.wsi$casecon[!missing.geno], site=as.numeric(factor(tp53epi.wsi$site[!missing.geno])), BC = as.numeric(tp53epi.wsi$prev.BC[!missing.geno]), Age = tp53epi.wsi$refage[!missing.geno], p53 = p53.snp[!missing.geno])
  
  p53.df = data.frame(p53.data)
  write.csv(p53.df, file=paste(snp.name, ".csv", sep=""))
  if(length(site.names)>1){
    p53.full = glm(CaseCon ~ factor(site) + factor(site)*p53 + factor(site)*Age + BC, data=p53.df, family=binomial, x=T)
    p53.pooled = glm(CaseCon ~ factor(site) + p53 + factor(site)*Age + BC, data=p53.df, family=binomial)
    p53.null = glm(CaseCon ~ factor(site) +  factor(site)*Age + BC, data=p53.df, family=binomial)
    
    p53.df$Site = factor(p53.df$site)
    
    p53.me = glmer(CaseCon ~ BC + p53 + Age + (1|site)  + (0 + p53 | site) + (0 + Age | site), start=c(site=1.0, p53=.1, Age=.01),nAGQ=1 , data=p53.df, family=binomial)
    
    test=anova(p53.full, p53.pooled, p53.null, test="Chi")
    coef = summary(p53.full)$coef
    ns = length(site.names)
    OR = coef[c(ns+1, (ns+4):(ns+ns+2)),1]
    x = p53.full$x
    p = predict(p53.full, type="response")
    var = solve(t(x)%*% diag(p*(1-p)) %*%x)[c(ns+1, (ns+4):(ns+ns+2)),c(ns+1, (ns+4):(ns+ns+2))]
    
    sqrt(diag(var))
    
    eff = matrix(0, ns,ns)
    eff[,1] = 1
    for (i in 2:ns) eff[i,i] = 1
    
    OR = eff %*% OR
    OR.SE = sqrt(diag(eff %*% var %*% t(eff)))
    
    DS = meta.summaries(OR, OR.SE, method="random", names=site.names, logscale=F)
    p.value = pnorm(-(abs(DS$summary/DS$se.summary)))*2
    BF0 = -exp(1)*p.value*log(p.value)
    return(list(snp=snp.name, DS=DS, OR=OR, SE=OR.SE, p.value=p.value, BF.Ha = 1/BF0, test=test, p53.me=p53.me))}
  else{
    p53.full = glm(CaseCon ~ p53 + Age + BC, data=p53.df, family=binomial, x=T)
    p53.null = glm(CaseCon ~ Age + BC, data=p53.df, family=binomial)
    test=anova(p53.full, p53.null, test="Chi")
    coef = summary(p53.full)$coef
    ns = length(site.names)
    OR = as.matrix(coef[2,1])
    OR.SE = (coef[2,2])
    p.value = coef[2,4]
    BF0 = -exp(1)*p.value*log(p.value)
    return(list(snp=snp.name, DS=DS, OR=OR, SE=OR.SE, p.value=p.value, BF.Ha = 1/BF0))
  }
}

```

```{r actual freq analysis}
#validation.sitenames = c("AUS" ,"HAW" ,"MAL" ,"NEC" ,"NHS" ,"SEA" ,"STA" ,"UCI","UKO" ,"USC")

freq<-OR.freq(snp.name, psdir="ps", drop.sites=NULL,iter=5000, debug=F) 

d = data.frame(t(freq$OR),coef(summary(freq$p53.me))["p53","Estimate"])
colnames(d) <- c(freq$DS$names, "Overall")
kable(t((d)))
```

- put this into a table-- see paper

```{r}
freq$p.value
```

- p values,
```{r}

```

fdr, BF interpretations
### model(s)

- rationale for priors
- plot for mixture
- plots of cond likelihood of mu

```{r more data stuff}
# p53.data.aug<-p53.data
# p53.data.aug$zeroes<- rep(0,p53.data$n.sites)
# p53.data.aug$zero<- 0

p53.data.normal<-p53.dataval
p53.data.normal$n.discovery<- length(p53.data.normal$discovery.sites)
p53.data.normal$zeroes<- rep(0,p53.data.normal$n.discovery)
p53.data.normal$MLE<- freq$OR[,1] 
p53.data.normal$SE<- freq$SE 
p = 0.00325
p53.data.normal$q<-qnorm(1-p/2)

bfdata <- p53.dataval
bfdata$p <- freq$p.value
```


```{r original}

p53.model = function() {
  for (j in 1:J) {
    CaseCon[j] ~ dbern(theta[j])
    logit(theta[j]) <- beta.site[site[j]] + beta.p53[site[j]]*p53[j] +
      beta.Age[site[j]]*Age[j] + beta.BC*BC[j]
  }
  for (k in 1:n.sites) {
    beta.site[k] ~ dnorm(mu.site, phi.site)
    beta.p53[k] ~ dnorm(mu.p53, phi.p53)
    beta.Age[k] ~ dnorm(mu.Age, phi.Age)
  }
  
  beta.BC ~ dnorm(0, 3)
  
  mu.site ~ dnorm(0, .1)
  phi.site <- pow(sigma.site, -2)
  sigma.site ~ dunif(0,5)
  
  mu.p53 ~ dnorm(0, .1)
  phi.p53 <- pow(sigma.p53, -2)
  sigma.p53 ~ dunif(0, 5)
  
  mu.Age ~ dnorm(0, .1)
  phi.Age <- pow(sigma.Age, -2)
  sigma.Age  ~ dunif(0, 5)
  
}

```



```{r latent var}
p53.newmodel = function() {
  for (j in 1:J) {
    CaseCon[j] ~ dbern(theta[j])
    logit(theta[j]) <- beta.site[site[j]] + beta.p53[site[j]]*p53[j] +
      beta.Age[site[j]]*Age[j] + beta.BC*BC[j]    }
  
  for (l in 1:n.sites) {
    beta.site[l] ~ dnorm(mu.site, phi.site)
    beta.p53.1[l] ~ dnorm(mu.p53, phi.p53)
    beta.p53[l] <- beta.p53.1[l]*assoc
    beta.Age[l] ~ dnorm(mu.Age, phi.Age)
  }
  beta.BC ~ dnorm(0, .1)
  
  mu.site ~ dnorm(0, .1)
  phi.site ~ dgamma(1,.05)
  sigma.site <- pow(phi.site, -.5)
  
  #E[prec] 20
  #(based on range .5 to 2 for OR => range = 1.4 = 6 sigma  sigma = 1.4/6 ~= .2 
  mu.p53<- mu1.p53*assoc
  mu1.p53 ~ dt(0,.1, 1)
  phi.p53 <- pow(sigma.p53, -2)
  sigma.p53 ~ dt(0,1,1)%_%T(0,)
  
  mu.Age ~ dnorm(0, .1)
  phi.Age ~ dgamma(1, .05)
  sigma.Age  <- pow(phi.Age, -.5)
  
  assoc ~ dbern(pind) 
  pind ~ dbeta(.5,.5)
}
```


```{r cond likelihood}

#mixed effect conditional likelihood
p53.normal = function() {
  for (j in 1:J) {
    CaseCon[j] ~ dbern(theta[j])
    logit(theta[j]) <- beta.site[site[j]] + beta.p53[site[j]]*p53[j] +
      beta.Age[site[j]]*Age[j] + beta.BC*BC[j]    }
  
  for (l in 1:n.sites) {
    beta.site[l] ~ dnorm(mu.site, phi.site)
    beta.p53[l] ~ dnorm(mu.p53, phi.p53)
    beta.Age[l] ~ dnorm(mu.Age, phi.Age)
  }
  C<-1000
  
  for(m in (n.sites+1):(n.sites+n.discovery)){
    beta.p53[m] ~ dnorm(mu.p53, phi.p53)
  }
  
  for (k in 1:n.discovery){
    #zeroes trick for MLE~cond prob
    tau[k]<- pow(SE[k], -2)
    L[k]<- dnorm(MLE[k],beta.p53[k+n.sites], tau[k])/
      (pnorm(-q*SE[k], beta.p53[k+n.sites], tau[k]) +
         1-pnorm(q*SE[k], beta.p53[k+n.sites], tau[k]))
    phi[k]<- -log(L[k])+C
    zeroes[k]~dpois(phi[k])
  }
  
  beta.BC ~ dnorm(0, .1)
  
  mu.site ~ dnorm(0, .1)
  phi.site ~ dgamma(1,.05)
  sigma.site <- pow(phi.site, -.5)
  
  #E[prec] 20
  #(based on range .5 to 2 for OR => range = 1.4 = 6 sigma  sigma = 1.4/6 ~= .2 
  mu.p53 ~ dt(0,.1,1)
  phi.p53 <- pow(sigma.p53, -2)
  sigma.p53 ~ dt(0,1,1)%_%T(0,)
  
  mu.Age ~ dnorm(0, .1)
  phi.Age ~ dgamma(1, .05)
  sigma.Age  <- pow(phi.Age, -.5)
  #assoc~dbern(pind) *assoc
  pind ~ dbeta(.5,.5)
}

```

```{r bf approx}

#eplogp approximation of BF for posterior prob of association
p53.bf.approx = function() {
  for (j in 1:J) {
    CaseCon[j] ~ dbern(theta[j])
    logit(theta[j]) <- beta.site[site[j]] + beta.p53[site[j]]*p53[j] +
      beta.Age[site[j]]*Age[j] + beta.BC*BC[j]    }
  
  for (l in 1:n.sites) {
    beta.site[l] ~ dnorm(mu.site, phi.site)
    beta.p53.1[l] ~ dnorm(mu.p53, phi.p53)
    beta.p53[l] <- beta.p53.1[l]*(assoc)
    beta.Age[l] ~ dnorm(mu.Age, phi.Age)
  }
  beta.BC ~ dnorm(0, .1)
  mu.site ~ dnorm(0, .1)
  phi.site ~ dgamma(1,.05)
  sigma.site <- pow(phi.site, -.5)
  
  #E[prec] 20
  #(based on range .5 to 2 for OR => range = 1.4 = 6 sigma  sigma = 1.4/6 ~= .2 
  mu.p53.1 ~ dt(0,.1,1)
  mu.p53<-mu.p53.1*assoc
  phi.p53 <- pow(sigma.p53, -2)
  sigma.p53 ~ dt(0,1,1)%_%T(0,)
  
  mu.Age ~ dnorm(0, .1)
  phi.Age ~ dgamma(1, .05)
  sigma.Age  <- pow(phi.Age, -.5)
  
  assoc~dbern(post.ind)
  prior.odds<- (pind)/(1-pind) #a/0
  BF<- 1/(-exp(1)*p*log(p)) #eplogp is BF h0/h1
  post.ind<-prior.odds*BF/(1+prior.odds*BF)
  pind ~ dbeta(.9,.9)
}
```


```{r run models}
parameters = c("mu1.site", "mu1.p53",  "beta.BC", "beta.Age", "mu.site","mu.p53", "mu.Age", "sigma.site", "sigma.p53",  "sigma.Age" ,"assoc", "pind", "phi1.site","phi1.p53","beta.site", "beta.p53","phi.site","phi.p53")

parameters.normal<- c("beta.BC", "beta.Age", "mu.site","mu.p53", "mu.Age", "sigma.site", "sigma.p53",  "sigma.Age" , "pind", "beta.site", "beta.p53", "assoc")

p53.sim = jags(data=p53.data, inits=NULL, parameters.to.save =parameters, model = p53.model)
p53.simval = jags(data=p53.dataval, inits=NULL, parameters.to.save =parameters, model = p53.model)
p53.simnew2 = jags(data=p53.data , inits=NULL, parameters.to.save =parameters, model = p53.newmodel) #data aug?

```

```{r}
p53.simnormal = jags(data=p53.data.normal, inits=NULL, parameters.to.save =parameters.normal, model = p53.normal)
p53.bfsim = jags(data=bfdata, inits=function(){list("pind"=.9)}, parameters.to.save =parameters, model = p53.bf.approx)
```




```{r comparison}
getOR<-function(sim, totalsites,discoverysites){
  
  OR = as.data.frame(sim$BUGSoutput$sims.matrix)%>%select( starts_with("beta.p53"), "mu.p53")
  exp(OR)
  #colnames(OR) = c(site.names, "Overall")
  
  sum.OR = t(apply(exp(OR), 2, function(x) {PI = HPDM(as.matrix(x),pointmass=1)
  return(c(median(x), PI[1], PI[2], PI[3],PI[4]))} #how to deal w multi, change HPD to have point mass at 1
  ))
  if (dim(sum.OR)[1]!= totalsites+1){
    sum.OR1<- matrix(NA, totalsites+1, dim(sum.OR)[2])
    sum.OR1[-discoverysites,]<- sum.OR
    sum.OR<- sum.OR1
  } ###
  #colnames(sum.OR)<-c("Median","2.5%","97.5%")
  return(sum.OR)
}

OR1<-getOR(p53.sim) #regular
ORval<-getOR(p53.simval)
ORnew<-getOR(p53.simnew) #w assoc
ORnormal<-getOR(p53.simnormal) #MLEs for prior
ORbf<-getOR(p53.bfsim) #MLEs for prior

```

```{r}
#ORtable = data.frame(original= OR1["mu.p53",], validation = ORval["mu.p53",], latentvar= ORnew["mu.p53",], cond= ORnormal["mu.p53",], bfapprox= ORbf["mu.p53",])
ORtable = data.frame(original= OR1, validation = ORval, latentvar= ORnew, cond= ORnormal, bfapprox= ORbf)

kable(ORtable)
simlist<- list(p53.sim,p53.simnew,p53.simnormal)
modelnames<- c("Original","Bayesian", "Conditional Likelihood")
plots<-function(simlist){
  ORlist = sapply(simlist, function(sim){ 
    OR<-sim$BUGSoutput$sims.list$mu.p53
    return(exp(OR))
  }
  )
  rownames(ORlist)<-NULL
  colnames(ORlist)<-modelnames
  plotdat<-melt(ORlist)
  plt<-ggplot()+geom_boxplot(data = plotdat, outlier.shape = NA,aes(x=Var2, y=value, group=factor(Var2)))+  theme_minimal() + scale_colour_few()
  #  + scale_y_continuous(limits = c(lower, upper))
  return(plt)
}

ggp<-plots(simlist)
ggp #just overall

```


```{r}

plotvar(p53.simnew$BUGSoutput$sims.list$mu.p53)
plotvar(p53.simnormal$BUGSoutput$sims.list$mu.p53)
plotvar(p53.bfsim$BUGSoutput$sims.list$mu.p53)
```






```{r all together}
getalldata<- function(iter=5000, drop.sites= NULL){
  
  #more iterations, coda traceplots, maybe look at the prior still- cauchy prior? ideally would do joint update (check if model is reducible-> chain gets stuck)-> maube google this bc jags probably 
  #
  
  load("tp53.Rdata")
  
  if(!is.null(drop.sites)) {
    use = !(tp53epi.wsi$site %in% drop.sites)
  } else {
    use = rep(TRUE, nrow(tp53epi.wsi))
  }
  
  names = colnames(tp53geno.wsi)
  maxsite = 0
  CaseCon<- site<-  BC <-  Age <- SNP <- p53<-NULL
  for(i in 1:length(names)){
    snp.name<- names[i]
    p53.snp = tp53geno.wsi[use, snp.name]
    tp53epi.wsi = tp53epi.wsi[use,]
    
    missing.geno = is.na(p53.snp)
    site.names = levels(factor(tp53epi.wsi[!missing.geno,"site"]))
    
    CaseCon<- c(CaseCon, tp53epi.wsi$casecon[!missing.geno])
    site<-c(site, as.numeric(factor(tp53epi.wsi$site[!missing.geno]))+
              maxsite)
    maxsite <-max(site)
    BC <- c(BC, as.numeric(tp53epi.wsi$prev.BC[!missing.geno]))
    Age <- c(Age, tp53epi.wsi$refage[!missing.geno])
    SNP <-c(SNP,  rep(i, length(unique(tp53epi.wsi$site[!missing.geno]))))
    p53<-c(p53, p53.snp[!missing.geno])
  }
  p53.data<- list(CaseCon = CaseCon,site=site,BC = BC,Age = Age,SNP = SNP,p53 = p53)
  J = length(p53.data$CaseCon)
  n.sites = length(unique(p53.data$site))
  p53.data$J = J
  p53.data$n.sites = n.sites
  p53.data$n.snp = length(unique(p53.data$SNP))
  return(p53.data)
}

p53.newmodel.all = function() {
  for (j in 1:J) {
    CaseCon[j] ~ dbern(theta[j])
    logit(theta[j]) <- beta.site[site[j]] + beta.p53[site[j]]*p53[j] +
      beta.Age[site[j]]*Age[j] + beta.BC[SNP[site[j]]]*BC[j]    }
  
  for (l in 1:n.sites) {
    beta.site[l] ~ dnorm(mu.site[SNP[l]], phi.site[SNP[l]])
    beta.p53.1[l] ~ dnorm(mu.p53[SNP[l]], phi.p53[SNP[l]])
    beta.p53[l] <- beta.p53.1[l]*assoc[SNP[l]]
    beta.Age[l] ~ dnorm(mu.Age[SNP[l]], phi.Age[SNP[l]])
  }
  
  for (k in 1:n.snp){
    beta.BC[k] ~ dnorm(0, .1)
    mu.site[k] ~ dnorm(0, .1)
    phi.site[k] ~ dgamma(1,.05)
    sigma.site[k] <- pow(phi.site[k], -.5)
    
    #E[prec] 20
    #(based on range .5 to 2 for OR => range = 1.4 = 6 sigma  sigma = 1.4/6 ~= .2 
    mu.p53[k]<- mu1.p53[k]*assoc[k]
    mu1.p53[k] ~ dt(0,.1, 1)
    phi.p53[k] <- pow(sigma.p53[k], -2)
    sigma.p53[k] ~ dt(0,1,1)%_%T(0,)
    
    mu.Age[k] ~ dnorm(0, .1)
    phi.Age[k] ~ dgamma(1, .05)
    sigma.Age[k]  <- pow(phi.Age[k], -.5)
    
    assoc[k] ~ dbern(pind[k]) 
    pind[k] ~ dbeta(.5,.5)
  }
  
}


p53.simnew.all = jags(data=getalldata(), inits=NULL, parameters.to.save =parameters, model = p53.newmodel.all)

```


```{r}
MU.ALL<-as.data.frame(p53.simnew.all$BUGSoutput$sims.matrix)%>%select( starts_with("mu.p53"))
apply(MU.ALL,2,plotvar)
```


```{r}

plotvar(exp(MU.ALL[,which(colnames(tp53geno.wsi)==snp.name)]),pointmass=1)
HPDM(as.matrix(exp(MU.ALL[,which(colnames(tp53geno.wsi)==snp.name)])),pointmass=1)
mean(exp(MU.ALL[,which(colnames(tp53geno.wsi)==snp.name)]))


```

```{r}

plotvar(exp(MU.ALL[,which(colnames(tp53geno.wsi)=="rs2287498n")]),pointmass=1)
HPDM(as.matrix(exp(MU.ALL[,which(colnames(tp53geno.wsi)=="rs2287498n")])),pointmass=1)
mean(exp(MU.ALL[,which(colnames(tp53geno.wsi)=="rs2287498n")]))
```


```{r}
plotvar(exp(p53.simnew$BUGSoutput$sims.list$mu.p53),pointmass=1)
HPDM(as.matrix(exp(p53.simnew$BUGSoutput$sims.list$mu.p53)),pointmass=1)
mean(exp(p53.simnew$BUGSoutput$sims.list$mu.p53))
```


- discussion of OR

The bayes factor and conditional likelihood 


- at least one example of a posterior?
- plot CI comparisons
- plot something like the shrinkage
- plot CI length comparison 
- some notes on disjoint CI's, log vs not logged HPD






```{r}
snp.name= snps[1]#"rs12951053n" 
discovery.sitenames= discovery.siteslist[[1]] #c("POL", "MAY", "NCO")


OR.runall<-function(snp.name, discovery.sitenames, iter=5000,p = 0.00325, psdir="ps"){
  #data
  
  p53.data<- getdata(snp.name, drop.sites=NULL)
  p53.dataval<- getdata(snp.name, drop.sites=discovery.sitenames)
  validation.sitenames = setdiff(p53.data$site.names, discovery.sitenames)
  freq<-OR.freq(snp.name, psdir="ps",
                drop=validation.sitenames,iter=5000, debug=F) 
  p53.data.normal<-p53.dataval
  p53.data.normal$n.discovery<- length(p53.data.normal$discovery.sites)
  p53.data.normal$zeroes<- rep(0,p53.data.normal$n.discovery)
  p53.data.normal$MLE<- freq$OR[,1] 
  p53.data.normal$SE<- freq$SE 
  p53.data.normal$q<-qnorm(1-p/2)
  bfdata <- p53.dataval
  bfdata$p <- freq$p.value
  
  #run all 
  parameters = c("beta.BC", "beta.Age", "mu.site","mu.p53", "mu.Age", "sigma.site", "sigma.p53",  "sigma.Age" ,"assoc", "pind","beta.site", "beta.p53","phi.site","phi.p53")
  
  p53.sim = jags(data=p53.data, inits=NULL, parameters.to.save =parameters, model = p53.model)
  p53.simnew = jags(data=p53.data , inits=NULL, parameters.to.save =parameters, model = p53.newmodel) 
  p53.simnormal = jags(data=p53.data.normal, inits=NULL, parameters.to.save =parameters.normal, model = p53.normal)
  p53.bfsim = jags(data=bfdata, inits=function(){list("pind"=.9)}, parameters.to.save =parameters, model = p53.bf.approx)
  
  #OR
  return(list(original= getOR(p53.sim, p53.data$n.sites, p53.data.normal$discovery.sites),
              bayes=getOR(p53.simnew, p53.data$n.sites, p53.data.normal$discovery.sites),
              cond=getOR(p53.simnormal, p53.data$n.sites, p53.data.normal$discovery.sites),
              bf=getOR(p53.bfsim, p53.data$n.sites, p53.data.normal$discovery.sites)))
}


```




```{r}

#test2<-OR.runall("rs2287498n",discovery.sitenames= discovery.sitenames)
#list of discovery sites
discovery.siteslist<-list(c("POL"),
                          c("POL", "MAY", "NCO"),
                          c("POL"),
                          c("POL"), 
                          c("NCO"),
                          c("POL", "MAY", "NCO"),
                          c("NCO")
                          
) 
snps<- setdiff(unique(colnames(tp53geno.wsi)), c( "rs2909430n", "rs2287499n", "rs2078486n"))
ORbySNP<- vector("list", length = length(snps))
for(i in 2:length(snps)){
  ORbySNP[[i]]<- OR.runall(snps[i],discovery.sitenames= discovery.siteslist[[i]])
  
}


```


```{r}
overall = do.call("rbind", lapply(ORbySNP, function(y) data.frame(t(sapply(y, function(x) (x[dim(x)[1],]))))))

#overall=data.frame(t(sapply(ORbySNP[[2]], function(x) x[dim(x)[1],])))
colnames(overall) = c("X","L","U", "L1","U1")
overall$model = rep(c("original","bayes","conditional","bf"),length(snps))
overall$snp = rep(snps, each = 4)
ggplot(data = overall, aes(group=model,color=model,x = snp,xend = snp))+ 
  geom_hline(yintercept=1, alpha=.3)+
  geom_linerange(aes(ymin=L, ymax=U), position = position_dodge(width = 1))+
  geom_linerange(aes(ymin=L1, ymax=U1), position = position_dodge(width = 1))+ 
  geom_point(aes(x=snp, y=X),size=3, shape=4,position = position_dodge(width = 1)) +
  coord_flip()+theme_minimal() + scale_colour_few() +
  
```

