<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>Bayesian Reproducibility</title>
  <meta name="description" content="Bayesian Reproducibility">
  <meta name="generator" content="bookdown 0.5 and GitBook 2.6.7">

  <meta property="og:title" content="Bayesian Reproducibility" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Bayesian Reproducibility" />
  
  
  

<meta name="author" content="Huijia Yu">



  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="8-conclusions-and-future-work.html">

<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />










</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./"></a></li>
<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Introduction</a></li>
<li class="chapter" data-level="1" data-path="1-abstract.html"><a href="1-abstract.html"><i class="fa fa-check"></i><b>1</b> Abstract</a></li>
<li class="chapter" data-level="2" data-path="2-lit-review.html"><a href="2-lit-review.html"><i class="fa fa-check"></i><b>2</b> Literature Review</a></li>
<li class="chapter" data-level="3" data-path="3-simulation-study.html"><a href="3-simulation-study.html"><i class="fa fa-check"></i><b>3</b> Simulation Study</a><ul>
<li class="chapter" data-level="3.1" data-path="3-simulation-study.html"><a href="3-simulation-study.html#normal-example"><i class="fa fa-check"></i><b>3.1</b> Normal Example</a><ul>
<li class="chapter" data-level="3.1.1" data-path="3-simulation-study.html"><a href="3-simulation-study.html#data-generation"><i class="fa fa-check"></i><b>3.1.1</b> Data Generation</a></li>
<li class="chapter" data-level="3.1.2" data-path="3-simulation-study.html"><a href="3-simulation-study.html#conditional-likelihood"><i class="fa fa-check"></i><b>3.1.2</b> Conditional Likelihood</a></li>
<li class="chapter" data-level="3.1.3" data-path="3-simulation-study.html"><a href="3-simulation-study.html#posterior-distribution"><i class="fa fa-check"></i><b>3.1.3</b> Posterior Distribution</a></li>
<li class="chapter" data-level="3.1.4" data-path="3-simulation-study.html"><a href="3-simulation-study.html#results"><i class="fa fa-check"></i><b>3.1.4</b> Results</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="4-tp53.html"><a href="4-tp53.html"><i class="fa fa-check"></i><b>4</b> TP53</a><ul>
<li class="chapter" data-level="4.1" data-path="4-tp53.html"><a href="4-tp53.html#data"><i class="fa fa-check"></i><b>4.1</b> Data</a></li>
<li class="chapter" data-level="4.2" data-path="4-tp53.html"><a href="4-tp53.html#previous-studies"><i class="fa fa-check"></i><b>4.2</b> Previous Studies</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="5-models.html"><a href="5-models.html"><i class="fa fa-check"></i><b>5</b> Models</a><ul>
<li class="chapter" data-level="5.1" data-path="5-models.html"><a href="5-models.html#fully-bayesian-model"><i class="fa fa-check"></i><b>5.1</b> Fully Bayesian Model</a></li>
<li class="chapter" data-level="5.2" data-path="5-models.html"><a href="5-models.html#conditional-likelihood-1"><i class="fa fa-check"></i><b>5.2</b> Conditional Likelihood</a></li>
<li class="chapter" data-level="5.3" data-path="5-models.html"><a href="5-models.html#bayes-factor-model"><i class="fa fa-check"></i><b>5.3</b> Bayes Factor Model</a></li>
<li class="chapter" data-level="5.4" data-path="5-models.html"><a href="5-models.html#methods"><i class="fa fa-check"></i><b>5.4</b> Methods</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="6-hierarchical-simulations.html"><a href="6-hierarchical-simulations.html"><i class="fa fa-check"></i><b>6</b> Hierarchical Simulations</a><ul>
<li class="chapter" data-level="6.1" data-path="6-hierarchical-simulations.html"><a href="6-hierarchical-simulations.html#data-generation-procedure"><i class="fa fa-check"></i><b>6.1</b> Data Generation Procedure</a><ul>
<li class="chapter" data-level="6.1.1" data-path="6-hierarchical-simulations.html"><a href="6-hierarchical-simulations.html#finding-discovery-sites"><i class="fa fa-check"></i><b>6.1.1</b> Finding “Discovery” Sites</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="6-hierarchical-simulations.html"><a href="6-hierarchical-simulations.html#results-1"><i class="fa fa-check"></i><b>6.2</b> Results</a><ul>
<li class="chapter" data-level="6.2.1" data-path="6-hierarchical-simulations.html"><a href="6-hierarchical-simulations.html#sensitivity"><i class="fa fa-check"></i><b>6.2.1</b> Sensitivity</a></li>
<li class="chapter" data-level="6.2.2" data-path="6-hierarchical-simulations.html"><a href="6-hierarchical-simulations.html#sensitivity-of-conditional-likelihood-method-to-changes-in-alpha"><i class="fa fa-check"></i><b>6.2.2</b> Sensitivity of Conditional Likelihood Method to Changes in <span class="math inline">\(\alpha\)</span></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="7-analysis-of-tp53-data.html"><a href="7-analysis-of-tp53-data.html"><i class="fa fa-check"></i><b>7</b> Analysis of TP53 Data</a></li>
<li class="chapter" data-level="8" data-path="8-conclusions-and-future-work.html"><a href="8-conclusions-and-future-work.html"><i class="fa fa-check"></i><b>8</b> Conclusions and Future Work</a><ul>
<li class="chapter" data-level="8.1" data-path="8-conclusions-and-future-work.html"><a href="8-conclusions-and-future-work.html#conclusion"><i class="fa fa-check"></i><b>8.1</b> Conclusion</a></li>
<li class="chapter" data-level="8.2" data-path="8-conclusions-and-future-work.html"><a href="8-conclusions-and-future-work.html#future-work"><i class="fa fa-check"></i><b>8.2</b> Future Work</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Bayesian Reproducibility</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="references" class="section level1 unnumbered">
<h1>References</h1>
<!--
This manually sets the header for this unnumbered chapter.
-->

<!--
To remove the indentation of the first entry.
-->

<!--
To create a hanging indent and spacing between entries.  These three lines may need to be removed for styles that don't require the hanging indent.
-->
<p>  </p>
<!--
This is just for testing with more citations for the bibliography at the end.  Add other entries into the list here if you'd like them to appear in the bibliography even if they weren't explicitly cited in the document.
-->

<div id="refs" class="references">
<div>
<p>Bacanu, S.-A., &amp; Kendler, K. S. (2013). Extracting actionable information from genome scans. <em>Genetic Epidemiology</em>, <em>37</em>(1), 48–59.</p>
</div>
<div>
<p>Benjamin, D. J., Berger, J. O., Johannesson, M., Nosek, B. A., Wagenmakers, E.-J., Berk, R., … others. (2017). Redefine statistical significance. <em>Nature Human Behaviour</em>.</p>
</div>
<div>
<p>Bigdeli, T. B., Lee, D., Webb, B. T., Riley, B. P., Vladimirov, V. I., Fanous, A. H., … Bacanu, S.-A. (2016). A simple yet accurate correction for winner’s curse can predict signals discovered in much larger genome scans. <em>Bioinformatics</em>, <em>32</em>(17), 2598–2603.</p>
</div>
<div>
<p>Ferguson, J. P., Cho, J. H., Yang, C., &amp; Zhao, H. (2013). Empirical bayes correction for the winner’s curse in genetic association studies. <em>Genetic Epidemiology</em>, <em>37</em>(1), 60–68.</p>
</div>
<div>
<p>Ghosh, A., Zou, F., &amp; Wright, F. A. (2008). Estimating odds ratios in genome scans: An approximate conditional likelihood approach. <em>The American Journal of Human Genetics</em>, <em>82</em>(5), 1064–1074.</p>
</div>
<div>
<p>Jiang, W., &amp; Yu, W. (2016). Power estimation and sample size determination for replication studies of genome-wide association studies. <em>BMC Genomics</em>, <em>17</em>(1), 19.</p>
</div>
<div>
<p>Phelan, C. M., Kuchenbaecker, K. B., Tyrer, J. P., Kar, S. P., Lawrenson, K., Winham, S. J., … others. (2017). Identification of 12 new susceptibility loci for different histotypes of epithelial ovarian cancer. <em>Nature Genetics</em>, <em>49</em>(5), 680.</p>
</div>
<div>
<p>Schildkraut, J. M., Goode, E. L., Clyde, M. A., Iversen, E. S., Moorman, P. G., Berchuck, A., … Garcia-Closas, M. (2009). Single nucleotide polymorphisms in the tp53 region and susceptibility to invasive epithelial ovarian cancer. <em>Cancer Research</em>, <em>69</em>(6), 2349–2357. <a href="http://doi.org/10.1158/0008-5472.CAN-08-2902" class="uri">http://doi.org/10.1158/0008-5472.CAN-08-2902</a></p>
</div>
<div>
<p>Schildkraut, J. M., Iversen, E. S., Wilson, M. A., Clyde, M. A., Moorman, P. G., Palmieri, R. T., … Berchuck, A. (2010). Association between dna damage response and repair genes and risk of invasive serous ovarian cancer. <em>PLoS One</em>, <em>5</em>(4), e10061.</p>
</div>
<div>
<p>Sellke, T., Bayarri, M., &amp; Berger, J. O. (2001). Calibration of <span class="math inline">\(\rho\)</span> values for testing precise null hypotheses. <em>The American Statistician</em>, <em>55</em>(1), 62–71.</p>
</div>
<div>
<p>Sun, L., Dimitromanolakis, A., Faye, L. L., Paterson, A. D., Waggott, D., Bull, S. B., … others. (2011). BR-squared: A practical solution to the winner’s curse in genome-wide scans. <em>Human Genetics</em>, <em>129</em>(5), 545–552.</p>
</div>
<div>
<p>Xu, L., Craiu, R. V., &amp; Sun, L. (2011). Bayesian methods to overcome the winner’s curse in genetic studies. <em>The Annals of Applied Statistics</em>, <em>5</em>, 201–231.</p>
</div>
<div>
<p>Xu, S. (2003). Theoretical basis of the beavis effect. <em>Genetics</em>, <em>165</em>(4), 2259–2268.</p>
</div>
<div>
<p>Yekutieli, D. (2012). Adjusted bayesian inference for selected parameters. <em>Journal of the Royal Statistical Society: Series B (Statistical Methodology)</em>, <em>74</em>(3), 515–541.</p>
</div>
<div>
<p>Zhong, H., &amp; Prentice, R. L. (2008). Bias-reduced estimators and confidence intervals for odds ratios in genome-wide association studies. <em>Biostatistics</em>, <em>9</em>(4), 621–634.</p>
</div>
<div>
<p>Zöllner, S., &amp; Pritchard, J. K. (2007). Overcoming the winner’s curse: Estimating penetrance parameters from case-control data. <em>The American Journal of Human Genetics</em>, <em>80</em>(4), 605–615.</p>
</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="8-conclusions-and-future-work.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": [["thesis.pdf", "PDF"], ["thesis.epub", "EPUB"], ["thesis.docx", "Word"]],
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(script.src))
      script.src  = script.src.replace(/^https?:/, '');
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
